src/clients/HubPoolClient.ts

```diff
@@ -428,14 +426,18 @@ export class HubPoolClient extends BaseAbstractClient {
 
     // Helper compute the realizedLpFeePct of an individual deposit based on pre-retrieved batch data.
     const computeRealizedLpFeePct = async (deposit: (typeof deposits)[0]) => {
-      const { originChainId, destinationChainId, inputToken, inputAmount, quoteTimestamp } = deposit;
+      const { originChainId, paymentChainId, inputToken, inputAmount, quoteTimestamp } = deposit;
       const quoteBlock = quoteBlocks[quoteTimestamp];
 
+      if (paymentChainId === undefined) {
```

Comment:
@pxrl I added this here so that the SpokePoolClient doesn't needlessly compute a realizedLpFeePct for V3 deposit events